### ISSUE TITLE
Mapper crashed on closing a map

### ISSUE THREAD

**Issue by sikmir** [OpenOrienteering/mapper#964]
### Steps to reproduce
1. Open map
2. Make changes, don't save
3. File->Close, Discard changes

### Actual behaviour
Program terminated with signal SIGSEGV, Address boundary error. Not stably reproducable.
```bash
TranslationUtil: Using qt_ru from /home/sikmir/projects/OpenOrienteering/mapper/cmake-build-debug/translations
TranslationUtil: Using OpenOrienteering_ru from /home/sikmir/projects/OpenOrienteering/mapper/cmake-build-debug/translations
TranslationUtil: Using map_symbols_ru from /home/sikmir/projects/OpenOrienteering/mapper/cmake-build-debug/translations
/home/sikmir/projects/OpenOrienteering/mapper/src/tools/tool.cpp:205:23: runtime error: member call on address 0x55ccb5380530 which does not point to an object of type 'MapEditorController'
0x55ccb5380530: note: object has a possibly invalid vptr: abs(offset to top) too big
 00 00 00 00  00 eb 60 b6 cc 55 00 00  d0 34 64 b6 cc 55 00 00  00 eb 60 b6 cc 55 00 00  d0 34 64 b6
              ^~~~~~~~~~~~~~~~~~~~~~~
              possibly invalid vptr
/home/sikmir/projects/OpenOrienteering/mapper/src/gui/map/map_editor.h:184:37: runtime error: member access within address 0x55ccb5380530 which does not point to an object of type 'MapEditorController'
0x55ccb5380530: note: object has a possibly invalid vptr: abs(offset to top) too big
 00 00 00 00  00 eb 60 b6 cc 55 00 00  d0 34 64 b6 cc 55 00 00  00 eb 60 b6 cc 55 00 00  d0 34 64 b6
              ^~~~~~~~~~~~~~~~~~~~~~~
              possibly invalid vptr
/home/sikmir/projects/OpenOrienteering/mapper/src/tools/tool_base.cpp:327:29: runtime error: member call on address 0x55ccb66434d0 which does not point to an object of type 'Map'
0x55ccb66434d0: note: object has a possibly invalid vptr: abs(offset to top) too big
 00 00 00 00  50 40 64 b6 cc 55 00 00  e1 04 00 00 00 00 00 00  20 05 38 b5 cc 55 00 00  50 ef 5c b6
              ^~~~~~~~~~~~~~~~~~~~~~~
              possibly invalid vptr
/usr/include/c++/7.2.0/bits/stl_tree.h:357:16: runtime error: member call on null pointer of type 'const struct _Rb_tree_node'
fish: './Mapper' terminated by signal SIGSEGV (Address boundary error)
```

### Configuration
Mapper Version: latest master (https://github.com/OpenOrienteering/mapper/commit/9da31324046c75b3f6c4fabefd268ce492f641aa)
Operating System: Arch Linux

**Comment by dg0yt** [comment]
Looks like a debug build with sanitizers enabled, catching a double delete.

**Comment by dg0yt** [comment]
Which tool were you using before closing the map?

**Comment by sikmir** [comment]
I was just adding Control number (704) symbol onto the map, nothing else.

**Comment by sikmir** [comment]
Not reproduced anymore.

**Comment by dl3sdo** [comment]
This might have been the same crash as reported and solved by #1418
